### PR TITLE
rspc 0.1.3 prep

### PR DESCRIPTION
**Pull request by oscartbeaumont**
Changes:
 - Change `rspc::Type` to `specta::Type` ready for the upstream change over
 - Upgrade rspc
 - Upgrade Specta
 - Upgrade PCR
 - Library subscriptions that don't suck finally!!!!!
 - Also removed all custom `impl Type`'s which is gonna be nice for easy Specta upgrades even over breaking changes.
 - The library query system's ease of maintenance has regressed in this PR but it's gonna be redone soon anyway.

### CODE REVIEW

**Comment by oscartbeaumont** [comment]
This PR also fixes this which is an edge case I didn't consider in the error logging. Everything is still working but these logs are stupid annoying.
image

File: core/src/api/sync.rs [hunk 1]
```diff
@@ -5,9 +5,9 @@ use super::{utils::LibraryRequest, RouterBuilder};
 pub fn mount() -> RouterBuilder {
 	RouterBuilder::new()
 		.library_subscription("newMessage", |t| {
-			t(|ctx, _: (), library_id| {
+			t(|ctx, _: (), library| {
 				async_stream::stream! {
-					let Some(lib) = ctx.library_manager.get_ctx(library_id).await else {
+					let Some(lib) = ctx.library_manager.get_ctx(library.id).await else {
```
Comment (Brendonovich) [inline core/src/api/sync.rs:10]: Isn't this unnecessary since we already have `library`?

File: crates/crypto/Cargo.toml [hunk 2]
```diff
@@ -8,7 +8,8 @@ edition = "2021"
 rust-version = "1.67.0"
 
 [features]
-rspc = ["dep:rspc"]
+rspc = ["dep:rspc", "dep:specta"]
```
Comment (Brendonovich) [inline crates/crypto/Cargo.toml:11]: i think only the `specta` feature is needed now

Comment (oscartbeaumont) [reply crates/crypto/Cargo.toml:11]: Nope. rspc is for `rspc::Error`. They can probs be one feature but I am keeping the previous semantics.

Comment (Brendonovich) [reply crates/crypto/Cargo.toml:11]: Ah ok cool

**Review by Brendonovich — APPROVED** [review]

**Pull request merged**